spec/content/index_data/manifest.spec.yml

```diff
@@ -0,0 +1,69 @@
+##
+## Describes the specification for a Kibana knowledge base entry's manifest.yml file
+##
+spec:
+  type: object
+  additionalProperties: false
+  properties:
+    type:
+      type: string
+      description: >
+        Type of the index data.
+      enum:
+        - knowledge_base_entry
+    description:
+      description: A description of what the index data provides
+      type: string
+
+    # Conditional properties.
+    title: true
+    index: true
+    retrieval: true
+  allOf:
+    - if:
+        properties:
+          type:
+            const: knowledge_base_entry
+        required:
+          - type
+      then:
+        properties:
```

Comment:
This is the part I'm the less sure about. Are we fine with all subtype's fields "leaking" at top level, or would we rather have dedicated groups per subtype, e.g.

```
knowledge_base_entry:
   type: object
   properties: 
       [...]
```